zephyr/lib/regions_mm.c

```diff
@@ -715,3 +717,5 @@ struct vmh_heap *vmh_get_heap_by_attribute(uint32_t attr, uint32_t core_id)
 	}
 	return NULL;
 }
+
+#endif /* if defined (CONFIG_MM_DRV) */
```

Comment:
Don't put whole files under an #if.  Take it out of the build in CMakeLists.txt.  For kconfig predication there's even library support, c.f. zephyr_library_sources_ifdef().